ts/src/scenario.ts

```diff
@@ -89,27 +113,84 @@ export class Scenario {
     return conductor;
   }
 
+  /**
+   * Create conductors with agents and add them to the scenario.
+   *
+   * The specified number of conductors is created and one agent is
+   * generated on each conductor.
+   *
+   * @param amount - The number of players to be created.
+   * @param networkConfig - Optional {@link NetworkConfig}
+   * @returns An array of {@link Player}s
+   */
+  async addPlayers(
+    amount: number,
+    networkConfig?: NetworkConfig,
+  ): Promise<Player[]> {
+    await this.ensureLocalServices();
+    return Promise.all(
+      new Array(amount).fill(0).map(async () => {
+        const conductor = await this.addConductor();
+        if (networkConfig) {
+          conductor.setNetworkConfig(networkConfig);
+        }
+        const agentPubKey = await conductor.adminWs().generateAgentPubKey();
+        return { conductor, agentPubKey };
+      }),
+    );
+  }
+
+  /**
+   * Installs the provided apps for the provided players.
+   *
+   * The number of players must be at least as high as the number of apps.
+   * **The agent pub key of the app options will be overwritten by the player's
+   * agent pub key**.
+   *
+   * @param appsWithOptions - The apps with options to be installed
+   * @param players - The players the apps are installed for
+   * @returns An array of player apps.
+   */
+  async installAppsForPlayers(
+    appsWithOptions: AppWithOptions[],
```

Comment:
I find the API here a bit confusing. My intuitive interpretation was that it would install all apps provided with the `appsWithOptions` argument for each player. Why not only accept a single `AppWithOptions` and install it for each of the players? If multiple apps need installing, the function could just be called multiple times.

Reply:
This is the same API as `addPlayersWithApps`, and I thought to just use the same app and options for all players. I hadn't done that because I was suspecting there might be a case where there are different options. Or different apps. But you're right, that's a special case and there are other ways of achieving that.
Any objections, @zo-el, @zippy? I'd apply that to this method and to `addPlayersWithApps` too.

Reply:
No objections from my side.

Reply:
Passing a different mem-proof for each agent is common for our apps.  And for that, I think we may definitely need to pass in an array of AppWithOptions .